docs/community/contribute/checks/service-support.md

```diff
@@ -0,0 +1,68 @@
+
+
+### Create a new file in the provider directory
+
+In this example, we are adding the CodeBuild service to the AWS provider.
+
+First create a new directory and file for your new service under the provider directory: e.g. [aws/codebuild/codebuild.go](https://github.com/aquasecurity/trivy/blob/main/pkg/iac/providers/aws/codebuild/codebuild.go)
+
+The CodeBuild service will require a structure `struct` to hold the information on input that is scanned. The input is the CodeBuild resource that a user configured and wants to scan for misconfiguration.
+
+
+```
+type CodeBuild struct {
+	Projects []Project
+}
+```
+
+The CodeBuild service manages `Project` resources. The `Project` struct has been added to hold information about each Project resources; `Project` Resources in turn manage `ArtifactSettings`:
+
+```
+type Project struct {
+	Metadata                  iacTypes.Metadata
+	ArtifactSettings          ArtifactSettings
+	SecondaryArtifactSettings []ArtifactSettings
+}
+
+type ArtifactSettings struct {
+	Metadata          iacTypes.Metadata
+	EncryptionEnabled iacTypes.BoolValue
+}
+```
+
+The `iacTypes.Metadata` struct is embedded in all of the Trivy types, and provides a common set of metadata for all resources. This includes the file and line number where the resource was defined, and the name of the resource.
+
+A resource, in this example `Project`, can have a name, and can optionally be encrypted. Instead of using raw string and bool types respectively, we use the trivy types `iacTypes.Metadata` and `iacTypes.BoolValue`. These types wrap the raw values and provide additional metadata about the value, such as whether it was set by the user or not, and the file and line number where the resource was defined. 
+
+Have a look at the other providers and services in the [`iac/providers`](https://github.com/aquasecurity/trivy/tree/main/pkg/iac/providers) directory in Trivy.
+
+Next you'll need to add a reference to your new service struct in the [provider struct](https://github.com/aquasecurity/trivy/blob/main/pkg/iac/providers/aws/aws.go) at `pkg/iac/providers/aws/aws.go`:
+
+```
+type AWS struct {
+	...
+	CodeBuild      codebuild.CodeBuild
+    ...
+}
+```
+
+Now you'll need to update all of the [adapters](https://github.com/aquasecurity/trivy/tree/main/pkg/iac/adapters) which populate the AWS provider struct. For example, if you want to support Terraform, you'll need to update [`trivy/pkg/iac/adapters/terraform/aws/codebuild/adapt.go`](https://github.com/aquasecurity/trivy/blob/main/pkg/iac/adapters/terraform/aws/codebuild/adapt.go).
+
+TODO HOW?
+
+## Create a new Schema for your provider
+
+Once the new service has been added to the provider, you need to create a the schema for the service as part of the provider schema. In the Trivy root directory run `make schema` to generate the schema for your new service.
+
+TODO
```

Comment:
We've updated to use `magefiles` as the rest of the Trivy targets. You can find them [here](https://github.com/aquasecurity/trivy/blob/main/magefiles/magefile.go#L429-L437)

So to generate the new schema it would be `mage schema:generate` and to verify it, the command will be `mage schema:verify`

You can see these targets by running `mage -l` in the Trivy repo.